File: examples/go/log.go
package main

import (
	"encoding/json"
	"fmt"
	"io"
)

// ReadyJSON is the shared CLI readiness payload.
func ReadyJSON() string {
	return `{"status":"ok"}`
}

// LogJSON returns one structured log line (no PII).
func LogJSON(level, msg string) string {
	raw, err := json.Marshal(map[string]string{"level": level, "msg": msg})
	if err != nil {
		return `{"level":"error","msg":"log"}`
	}
	return string(raw)
}

// Run is the CLI entry (stdout = user payload, stderr = JSON logs).
func Run(args []string, stdout, stderr io.Writer) int {
	for _, arg := range args {
		if arg == "--ready" {
			fmt.Fprintln(stderr, LogJSON("info", "ready"))
			fmt.Fprintln(stdout, ReadyJSON())
			return 0
		}
	}
	fmt.Fprintln(stderr, LogJSON("info", "start"))
	fmt.Fprintln(stdout, Greet())
	fmt.Fprintln(stdout, AboutSummary())
	return 0
}

File: examples/python/src/hello/log.py
"""Structured JSON logs and readiness (no PII)."""

from __future__ import annotations

import json
import sys


def ready_json() -> str:
    return json.dumps({"status": "ok"}, separators=(",", ":"))


def log_json(level: str, msg: str) -> str:
    return json.dumps({"level": level, "msg": msg}, separators=(",", ":"))


def log_event(level: str, msg: str) -> None:
    print(log_json(level, msg), file=sys.stderr)
